Framework/Core/include/Framework/AnalysisDataModel.h

```diff
@@ -1629,12 +1629,39 @@ namespace aod
 {
 namespace mctracklabel
 {
-DECLARE_SOA_INDEX_COLUMN(McParticle, mcParticle); //! MC particle
-DECLARE_SOA_COLUMN(McMask, mcMask, uint16_t);     //! Bit mask to indicate detector mismatches (bit ON means mismatch). Bit 0-6: mismatch at ITS layer. Bit 7-9: # of TPC mismatches in the ranges 0, 1, 2-3, 4-7, 8-15, 16-31, 32-63, >64. Bit 10: TRD, bit 11: TOF, bit 15: indicates negative label
+DECLARE_SOA_INDEX_COLUMN(McParticle, mcParticle);            //! MC particle
+DECLARE_SOA_COLUMN(McMask, mcMask, uint16_t);                //! Bit mask to indicate detector mismatches (bit ON means mismatch). Bit 0-6: mismatch at ITS layer. Bit 7-9: # of TPC mismatches in the ranges 0, 1, 2-3, 4-7, 8-15, 16-31, 32-63, >64. Bit 10: TRD, bit 11: TOF, bit 15: indicates negative label
+DECLARE_SOA_DYNAMIC_COLUMN(McMismatchInITS, mcMismatchInITS, //! Mismatch in the ITS layer
+                           [](uint16_t mcMask, int layer) -> bool {
+                             if (layer >= 7 || layer < 0) {
+                               return false;
+                             }
+                             return (mcMask & 1 << layer);
+                           });
+DECLARE_SOA_DYNAMIC_COLUMN(McMismatchInTPC, mcMismatchInTPC, //! Mismatch in the TPC
+                           [](uint16_t mcMask, int layer) -> bool {
+                             if (layer >= 10 || layer < 7) {
```

Comment:
What are these layers for the TPC? Shall we add an enum to DataTypes.h to make it more clear?

Reply:
@njacazio What do you think?